### PR TITLE
Add node upgrade logic for the controller

### PR DESCRIPTION
**Pull request by tsurai**
Implementation of a simple node image upgrade mechanism that rebuilds and restrarts the node container if a new image can be pulled independent of the used orchestration tool such as docker-compose.

### CODE REVIEW

**Review by solidnerd — COMMENTED** [review]
Is there a problem of starting two indy nodes on the same data dirs. Could it not be that the new container corrupt the indy data and than also the old container can't use the libindy data. Instead of spinning up the new container we should have a downtime in terms of shutting down the old and than starting the new one.

File: controller/upgrade_indy.sh [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    if [[ -z "$image_name" ]]; then
+        print_error "Container $container does not exist"
+        exit 22
+    fi
+
+    # save the creation date of the image
+    image_date="$($ENGINE image inspect "$image_name" --format '{{.Created}}')"
+
+    # try to pull a new image
+    output="$($ENGINE image pull "$image_name")"
```
[the remark this answers is not in the document]
Comment (tsurai) [reply controller/upgrade_indy.sh:45]: It is trivial to target another tag but right now we don't have any real convention of how we tag images hence only latest is being used. What we should have is both a mutable `latest` tag as is best practice for docker images as well as static versioned tags and have the upgrade pull the version specified in the upgrade transaction

**Comment by tsurai** [comment]
> Is there a problem of starting two indy nodes on the same data dirs. Could it not be that the new container corrupt the indy data and than also the old container can't use the libindy data. Instead of spinning up the new container we should have a downtime in terms of shutting down the old and than starting the new one.

That is the current behavior. The old container is being stopped properly and then a new one is started. They cannot run at the same time anyway due to both port and container name conflicts. A few seconds downtime cannot be avoided. 

The new container is only being _created_ but not actually run before the old one is stopped. This order reduces the necessary downtime and avoids stopping and removing the old container in case the new one fails to be created for whatever reason leaving us with no node at all.

**Comment by Echsecutor** [comment]
fixes https://github.com/IDunion/indy-node-container/issues/59

**Review by Echsecutor — APPROVED** [review]
as discussed last friday

**Pull request merged**